lib/enigma.rb

```diff
@@ -1,20 +1,18 @@
 require './lib/cipher'
 
 class Enigma < Cipher
-  def encrypt(message, key, date)
+  def encrypt(message, key = generate_key, date = @date)
     shifts = generate_shifts(key, date, 1)
-    mutate_message(message, shifts)
 
     {encryption: mutate_message(message, shifts),
      key: key,
      date: date}
   end
 
-  def decrypt(cipher_text, key, date)
+  def decrypt(cipher_text, key, date = @date)
     shifts = generate_shifts(key, date, -1)
-    mutate_message(cipher_text, shifts)
 
-    {encryption: mutate_message(cipher_text, shifts),
+    {decryption: mutate_message(cipher_text, shifts),
```

Comment:
This was at the point of the project where I started to get excited and stopped looking at details.